src/test/unit/controllers/blocks.controller.spec.ts

```diff
@@ -0,0 +1,318 @@
+import { HttpException, HttpStatus, INestApplication } from "@nestjs/common";
+import { mockBlockService } from "./services.mock/block.services.mock";
+import { Test, TestingModule } from "@nestjs/testing";
+import { BlockController } from "src/endpoints/blocks/block.controller";
+import { BlockService } from "src/endpoints/blocks/block.service";
+import request = require('supertest');
+import { PublicAppModule } from "src/public.app.module";
+import { QueryPagination } from "src/common/entities/query.pagination";
+
+describe("BlockController", () => {
+  let app: INestApplication;
+  const path = "/blocks";
+  const blockServiceMock = mockBlockService();
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      controllers: [BlockController],
+      imports: [PublicAppModule],
+    })
+      .overrideProvider(BlockService)
+      .useValue(blockServiceMock)
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+  });
+
+  describe('GET /blocks', () => {
+    it('should return a list of blocks', async () => {
+      const mockBlocksList = createMockBlocksList(10);
+      blockServiceMock.getBlocks.mockResolvedValue(mockBlocksList);
+
+      await request(app.getHttpServer())
+        .get(`${path}`)
+        .expect(200)
+        .expect(response => {
+          expect(response.body.length).toBe(10);
+          expect(response.body).toEqual(mockBlocksList);
+        });
+    });
+
+    it('should filter blocks by shard', async () => {
+      const shard = 1;
+      const filteredBlocksList = createMockBlocksList(5).map(block => ({ ...block, shard }));
+      blockServiceMock.getBlocks.mockResolvedValue(filteredBlocksList);
```

Comment:
can we check that the correct fields were passed to the service instead of manually preparing the expected data? 
for example: we just need to check that the service received the correct BlockFilter or QueryPagination.

Reply:
solved